From 9832f132423cdb21e19facd60db24892300df871 Mon Sep 17 00:00:00 2001
From: Purushottam Kushwaha <quic_pkushwah@quicinc.com>
Date: Fri, 7 Jun 2024 16:48:29 +0530
Subject: [PATCH] Add vendor flag to indicate unavailability mode in TWT
 responder mode

Add a flag attribute
QCA_WLAN_VENDOR_ATTR_TWT_SET_PARAM_UNAVAILABILITY_MODE into enum
qca_wlan_vendor_attr_twt_set_param to configure the TWT responder
unavailability outside of the SPs of its broadcast TWT schedule.

Signed-off-by: Purushottam Kushwaha <quic_pkushwah@quicinc.com>
---
 src/common/qca-vendor.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/common/qca-vendor.h b/src/common/qca-vendor.h
index 980e0167b..6ac31aa3e 100644
--- a/src/common/qca-vendor.h
+++ b/src/common/qca-vendor.h
@@ -11259,10 +11259,18 @@ enum qca_wlan_vendor_attr_twt_capability {
  * This attribute configures AC parameters to be used for all TWT
  * sessions in AP mode.
  * Uses the enum qca_wlan_ac_type values.
+ *
+ * @QCA_WLAN_VENDOR_ATTR_TWT_SET_PARAM_UNAVAILABILITY_MODE: Flag attribute,
+ * used by TWT responder to indicate unavailability outside of the SPs.
+ * Enable (flag attribute present) - Indicates that the TWT responder may be
+ * unavailable outside of the SPs of its broadcast TWT schedule.
+ * Disable (flag attribute not present) - Indicates that the responder will be
+ * available for all TWT sessions (including individual TWT).
  */
 enum qca_wlan_vendor_attr_twt_set_param {
 	QCA_WLAN_VENDOR_ATTR_TWT_SET_PARAM_INVALID = 0,
 	QCA_WLAN_VENDOR_ATTR_TWT_SET_PARAM_AP_AC_VALUE = 1,
+	QCA_WLAN_VENDOR_ATTR_TWT_SET_PARAM_UNAVAILABILITY_MODE = 2,
 
 	/* keep last */
 	QCA_WLAN_VENDOR_ATTR_TWT_SET_PARAM_AFTER_LAST,
-- 
2.43.0

